Tools/CISupport/ews-build/steps.py

```diff
@@ -7138,7 +7138,61 @@ class ParseStaticAnalyzerResultsWithoutChange(ParseStaticAnalyzerResults):
     scan_build_output = SCAN_BUILD_OUTPUT_DIR + '-baseline'
 
 
-class FindUnexpectedStaticAnalyzerResults(shell.ShellCommandNewStyle, AddToLogMixin):
+class FindModifiedSaferCPPExpectations(shell.ShellCommandNewStyle, AddToLogMixin):
+    name = 'find-modified-safer-cpp-expectations'
+    RE_FILE = r'^(\+|-)(?P<file>[^/+/-].+(?:\.cpp|\.mm|\.h))$'
+    RE_EXPECTATIONS = r'^(\+\+\+).+(Source/(?P<project>.+)/SaferCPPExpectations/(?P<checker>.+)Expectations)$'
+    command = ['git', 'diff', 'head~1', '--', '*Expectations']
+
+    def __init__(self, **kwargs):
+        super().__init__(logEnviron=False, **kwargs)
+
+    @defer.inlineCallbacks
+    def run(self):
+        self.log_observer = logobserver.BufferLogObserver()
+        self.addLogObserver('stdio', self.log_observer)
+
+        rc = yield super().run()
+        if rc != SUCCESS:
+            return defer.returnValue(rc)
+
+        logText = self.log_observer.getStdout()
+
+        if not logText:
+            yield self._addToLog('stdio', 'This change does not modify Safer CPP expectations.\n')
+            return defer.returnValue(rc)
+
+        yield self._addToLog('stdio', '\nLooking for changes to Safer CPP expectations...\n')
+        removed_tests = []
+        added_tests = []
+        for line in logText.splitlines():
+            expectation_match = re.search(self.RE_EXPECTATIONS, line, re.IGNORECASE)
+            if expectation_match:
+                project = expectation_match.group('project')
+                checker = expectation_match.group('checker')
+                yield self._addToLog('stdio', f'Changes for {project}/{checker}...\n')
+            file_match = re.search(self.RE_FILE, line, re.IGNORECASE)
+            if file_match:
+                test_name = f"{project}/{file_match.group('file')}/{checker}"
+                if file_match.group(1) == '+':
+                    added_tests.append(test_name)
+                    yield self._addToLog('stdio', f'    {test_name} was added to {checker} expectations.\n')
+                elif file_match.group(1) == '-':
+                    removed_tests.append(test_name)
+                    yield self._addToLog('stdio', f'    {test_name} was removed from {checker} expectations.\n')
+
+        self.setProperty('user_removed_tests', removed_tests)
+        self.setProperty('user_added_tests', added_tests)
+        return defer.returnValue(rc)
+
+    def getResultSummary(self):
```

Comment:
what about the case when the step result is : FAILURE, EXCEPTION, CANCELLED etc.